presto-native-execution/presto_cpp/main/dynamic_registry/examples/default_registrations/DynamicArrayFunction.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/dynamic_registry/DynamicFunctionRegistrar.h"
+#include "velox/type/SimpleFunctionApi.h"
+
+// This file defines a mock function that will be dynamically linked and
+// registered. There are no restrictions as to how the function needs to be
+// defined, but the library (.so) needs to provide a `void registry()` C
+// function in the top-level namespace.
+//
+// (note the extern "C" directive to prevent the compiler from mangling the
+// symbol name).
+
+namespace facebook::velox::common::dynamicRegistry {
+
+template <typename T>
+struct DynamicFunction {
```

Comment:
Nit : Change the name of this function to maybe DynamicArray so that it demonstrates that the struct name needn't be DynamicFunction only. 

Also, maybe combine the contents of 2 of these example files so that we end up having 2 example files, one with 2 functions and another with a single one.